cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -143,6 +145,16 @@ def index_item(item, skip_index=False):
             if not item_index_dictionary and not item.has_children:
                 return
 
+            if groups_usage_info:
+                item.content_groups = []
+                for name, group in groups_usage_info:
+                    for module in group:
+                        if str(item.location) == str(module['usage_key_string']):
+                            item.content_groups.append(name)
+
+            if content_groups and not getattr(item, 'content_groups', None):
+                item.content_groups = content_groups
```

Comment:
This seems like we're adding all content groups to the index for items that have no specification for content groups - I'm not sure I understand why this is necessary.